bytestring.cabal

```diff
@@ -113,9 +113,15 @@ library
                     -fmax-simplifier-iterations=10
                     -fdicts-cheap
                     -fspec-constr-count=6
+  if (arch(x86_64) || arch(i386))
+    c-sources:        cbits/fpstring.c
+                      cbits/itoa.c
+                      cbits/x86/is-valid-utf8.c
+  else
+    c-sources:        cbits/fpstring.c
+                      cbits/itoa.c
+                      cbits/is-valid-utf8.c
```

Comment:
Cabal file can declare multiple `c-sources`. So you can leave `fpstring.c` and `itoa.c` as is and add a conditional section to choose between `is-valid-utf8.c` only.

Reply:
Did not know this, thanks!